plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryPageSink.java

```diff
@@ -98,27 +111,39 @@ public CompletableFuture<?> appendPage(Page page)
             batch.put(row);
         }
 
-        insertWithCommitted(batch);
+        append(batch);
         return NOT_BLOCKED;
     }
 
-    private void insertWithCommitted(JSONArray batch)
+    private void append(JSONArray batch)
     {
-        try (JsonStreamWriter writer = JsonStreamWriter.newBuilder(writeStream.getName(), writeStream.getTableSchema(), client).build()) {
-            ApiFuture<AppendRowsResponse> future = writer.append(batch);
+        try {
+            ApiFuture<AppendRowsResponse> future = streamWriter.append(batch);
             AppendRowsResponse response = future.get(); // Throw error
             if (response.hasError()) {
                 throw new TrinoException(BIGQUERY_BAD_WRITE, format("Response has error: %s", response.getError().getMessage()));
             }
         }
-        catch (Exception e) {
-            throw new TrinoException(BIGQUERY_BAD_WRITE, "Failed to insert rows", e);
+        catch (IOException | DescriptorValidationException | ExecutionException | InterruptedException | AppendSerializationError e) {
+            throw new TrinoException(BIGQUERY_BAD_WRITE, "Failed to append stream writer", e);
```

Comment:
the message should start from saying what operation failed 
"Failed to append stream writer" may not be self-explanatory
What about using a simpler wording: `"Write failed: " + firstNonNull(e.getMessage(), e)` as a message?